src/Symfony/Component/Yaml/Parser.php

```diff
@@ -44,6 +48,16 @@ public function __construct($offset = 0, $totalNumberOfLines = null, array $skip
         $this->offset = $offset;
         $this->totalNumberOfLines = $totalNumberOfLines;
         $this->skippedLineNumbers = $skippedLineNumbers;
+        $this->tagResolver = new TagResolver();
+    }
+
+    /**
+     * @param TagInterface $tag
+     * @param int          $priority
+     */
+    public function addTag(TagInterface $tag, $priority = 0)
```

Comment:
It looked complicated to use to me as there are already 3 internal arguments in the constructor (that should be removed imo but not sure we can do that because of bc).
People would be forced to do `new Parser(0, null, array(), new TagResolver());`.